### ISSUE TITLE
Link will not embed a player when using direct .mp4 link

### ISSUE THREAD

**Issue by Snnackss** [Hubro/clipface#11]
Is it possible to get embed to work on something like Discord when using a direct link? As in direct link, I mean I am using the link that reads like this: https://MYWEBSITE.com:PORT/api/video/VIDEO.mp4

It would also be nice to get it to work with the "public link" url, but I am not sure if that would work or not.

This is what normally happens when I post a direct mp4 link (I am using Gyazo as an example): https://imgur.com/a/sIwv1WG

**Comment by Hubro** [comment]
Remember that if you have authentication enabled you can't link people directly to videos, since they are redirected to a login screen instead.

In any case I suspect Discord may respect some special HTML tags that identify the primary media element of the linked web page. I'll look into it.

**Comment by Snnackss** [comment]
> Remember that if you have authentication enabled you can't link people directly to videos, since they are redirected to a login screen instead.

What if you use the "public links"? If you were able to add special tags do you think it will still embed even if the .mp4 part of the URL is followed up with the public token? I know with Streamable they give you an embed even though its not a direct URL, but it also sucks because they lock it to 30FPS on top of all of their other free tier limitations.

**Comment by Hubro** [comment]
> > Remember that if you have authentication enabled you can't link people directly to videos, since they are redirected to a login screen instead.
> 
> What if you use the "public links"? If you were able to add special tags do you think it will still embed even if the .mp4 part of the URL is followed up with the public token? I know with Streamable they give you an embed even though its not a direct URL, but it also sucks because they lock it to 30FPS on top of all of their other free tier limitations.

Tried, nothing works. I noticed Streamable is using [OpenGraph](https://ogp.me) meta tags. I'll add those to Clipface and see if that works.

**Comment by Hubro** [comment]
@Snnackss This is now fixed. I have implemented OpenGraph video tags in Clipface, which Discord now picks up and uses to automatically embed the video. Just click "Copy link" or "Copy public link" if you have authentication enabled, and paste it into Discord.

NB: Discord will only play certain video formats. So far I have tested `video/x-matroska` (.mkv files) and `video/mp4` (.mp4 files), both X264 encoded, and only the MP4 files worked.
